tests/conftest.py

```diff
@@ -20,19 +22,78 @@
     'file-system',
 ]
 
+# UID for test session
+# Repositories on the host will be named
+# unittest-<session-uid>-<repository-uid>
+pytest.UID = audeer.uid()[:8]
+
+
+@pytest.fixture(scope='session', autouse=True)
+def cleanup_artifactory():
+    r"""Remove letover unit test repositories on Artifactory.
+
+    As removing a unit test repsoitory
+    in the ``backend()`` fixture
+    might fail
+    (https://github.com/audeering/audbackend/issues/97),
+    we try to clean up the host
+    everytime we run the tests.
+    If this fails
+    a ``RuntimeError`` is raised
+    and the user then needs to clean up
+    the repository manually.
+
+    """
+
+    yield
+
+    # Delete leftover repositories
+    name = 'artifactory'
+    host = pytest.HOSTS[name]
+    r = audfactory.rest_api_get(f'{host}/api/repositories')
+    if r.status_code == 200:
+        repos = [entry['key'] for entry in r.json()]
+        repos = [
+            repo for repo in repos
+            if repo.startswith(f'unittest-{pytest.UID}')
+        ]
+        remaining_repos = repos.copy()
+        for repo in repos:
+            try:
+                audbackend.delete(name, host, repo)
+                remaining_repos.remove(repo)
+            except audbackend.BackendError as ex:
+                error_msg = (
+                    f'Cleaning up of repo {repo} failed.\n'
+                    'Please delete remaining repositories manually with:\n'
+                )
+                for remaining_repo in remaining_repos:
+                    error_msg += (
+                        f"'audbackend.delete({name}, {host}, "
+                        f"{remaining_repo})',\n"
+                    )
+                error_msg += f'The original error message was {ex}.'
+                raise RuntimeError(error_msg)
```

Comment:
So the idea is that this let's the CI job fail and then the developer will look up which repositories were not deleted and do it manually. I wonder if it's simpler we raise the error directly in the backend fixture instead. Then it's always only one repository the developer has to delete and we don't need an additional `cleanup_artifactory()` fixture.